lib/chef-acceptance/application.rb

```diff
@@ -4,62 +4,91 @@
-        puts "#{e}\n#{e.backtrace.join("\n")}"
-      rescue RuntimeError
-        # We catch the errors here and do not raise again in
-        # order to make a clean exit. Since the errors are already
-        # in stdout, we do not print them again.
-        error = true
-      ensure
-        if run_summary
-          total_duration = Time.now - start_time
-          # Special footer row that gives overall status
-          output_formatter.add_row(suite: "", command: "Total", duration: total_duration, error: error)
-
-          puts ""
-          puts "chef-acceptance run #{error ? "failed" : "succeeded"}."
-          puts output_formatter.generate_output
+        suites = parse_suites(suite_regex)
+
+        puts "Running test suites:"
+        puts suites.join("\n")
+        puts ""
+
+        # Start the overall run timer
+        run_start_time = Time.now
+
+        suites.each do |suite|
+          suite_start_time = Time.now
+
+          begin
+            run_suite(suite, command)
+          rescue RuntimeError => e
+            # We catch the errors here and do not raise again in
+            # order to make a clean exit. Since the errors are already
+            # in stdout, we do not print them again.
+            errors[suite] = e
+          ensure
+            suite_duration = Time.now - suite_start_time
+            # Capture overall suite run duration
+            output_formatter.add_row(suite: suite, command: "Total", duration: suite_duration, error: errors[suite])
+          end
         end
+
+        total_duration = Time.now - run_start_time
+        # Special footer row that gives overall status for run
+        output_formatter.add_row(suite: "Run", command: "Total", duration: total_duration, error: run_error?)
+        print_summary
+      rescue AcceptanceFatalError => e
+        puts ""
+        puts e
+        exit(1)
       end
 
       # Make sure that exit code reflects the error if we have any
-      exit(1) if error
+      exit(1) if run_error?
+    end
+
+    def run_error?
+      !errors.empty?
+    end
+
+    # Parse the regex pattern and return an array of matching test suite names
+    def parse_suites(regex)
+      # Find all directories in cwd
+      suites = Dir.glob("*").select { |f| File.directory? f }
+      raise AcceptanceFatalError, "No test suites to run in #{Dir.pwd}" if suites.empty?
+
+      # Find all test suites that match the regex pattern
+      matched_suites = suites.select { |s| /#{regex}/ =~ s }
+      raise AcceptanceFatalError, "No matching test suites found using regex '#{regex}'" if matched_suites.empty?
+
+      matched_suites
     end
 
     def run_suite(suite, command)
       test_suite = TestSuite.new(suite)
 
       unless test_suite.exist?
-        raise AcceptanceError, <<-EOS.gsub(/^\s+/, "")
-          Could not find test suite '#{test_suite.name}' in the current working directory '#{Dir.pwd}'.
+        raise AcceptanceFatalError, <<-EOS.gsub(/^\s+/, "")
```

Comment:
We probably don't need this check anymore since this is validated by `parse_suites`.   I don't see a test for this condition in application_spec anyhow.